src/components/Image/ImageUriCache.js

```diff
@@ -0,0 +1,58 @@
+class ImageUriCache {
+  static _maximumItems: number = 256;
+  static _entries = {};
+
+  static has(uri: string) {
+    const entries = ImageUriCache._entries;
+    const isDataUri = /^data:/.test(uri);
+    return isDataUri || Boolean(entries[uri]);
+  }
+
+  static add(uri: string) {
+    const entries = ImageUriCache._entries;
+    if (entries[uri]) {
+      entries[uri].refCount += 1;
```

Comment:
Why not also update a `lastAccessed` timestamp? Small concern that as implemented, this isn't *least recently used*, but *first added*. Just because something doesn't exist on screen doesn't mean that it isn't a high priority and often used item (I'm thinking logged-in user avatar, for example)